agent/config/builder.go

```diff
@@ -387,6 +387,26 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		}
 	}
 
+	// expand dns recursors
+	uniq := map[string]bool{}
+	dnsRecursors := []string{}
+	for _, r := range c.DNSRecursors {
+		x, err := template.Parse(r)
+		if err != nil {
+			return RuntimeConfig{}, fmt.Errorf("Invalid DNS recursor template %q: %s", r, err)
+		}
+		for _, addr := range strings.Fields(x) {
+			if strings.HasPrefix(addr, "unix://") {
+				return RuntimeConfig{}, fmt.Errorf("DNS Recursors cannot be unix sockets: %s", addr)
+			}
```

Comment:
should this also check for invalid values like 0.0.0.0?

Reply:
This isn't something we've done before but is a useful addition.

Reply:
pushed a change to add that.